Sources/TuistAcceptanceTesting/TuistAcceptanceTestCase.swift

```diff
@@ -0,0 +1,141 @@
+import TSCBasic
+import TuistCore
+import TuistGraph
+import TuistSupportTesting
+import XCTest
+
+@testable import TuistKit
+@testable import TuistSupport
+
+public enum Destination {
+    case simulator, device
+}
+
+open class TuistAcceptanceTestCase: TuistTestCase {
+    public var xcodeprojPath: AbsolutePath!
+    public var workspacePath: AbsolutePath!
+    public var fixturePath: AbsolutePath!
+    public var derivedDataPath: AbsolutePath!
+    public var cacheDirectory: AbsolutePath!
+
+    private var sourceRootPath: AbsolutePath!
+
+    override open func setUp() {
+        super.setUp()
+
+        cacheDirectory = try! TemporaryDirectory(removeTreeOnDeinit: true).path
+        derivedDataPath = try! TemporaryDirectory(removeTreeOnDeinit: true).path
+        setenv(
+            Constants.EnvironmentVariables.forceConfigCacheDirectory,
+            cacheDirectory.pathString,
+            1
+        )
+
+        sourceRootPath = try! AbsolutePath(
+            validating: ProcessInfo.processInfo.environment[
+                "TUIST_CONFIG_SRCROOT"
+            ]!
+        )
+        environment.tuistConfigVariables[
+            Constants.EnvironmentVariables.xcbeautifyBinaryPath
+        ] = sourceRootPath
+            .appending(components: ["vendor", ".build", "debug", "xcbeautify"])
+            .pathString
+
+        DeveloperEnvironment.shared = DeveloperEnvironment()
+    }
+
+    override open func tearDown() async throws {
+        xcodeprojPath = nil
+        workspacePath = nil
+        fixturePath = nil
+        derivedDataPath = nil
+        cacheDirectory = nil
+
+        try await super.tearDown()
+    }
+
+    public func setUpFixture(_ fixture: String) throws {
+        let fixturesPath = sourceRootPath
+            .appending(component: "fixtures")
+
+        fixturePath = FileHandler.shared.currentPath.appending(component: fixture)
+
+        try FileHandler.shared.copy(
+            from: fixturesPath.appending(component: fixture),
+            to: fixturePath
+        )
+
+        try poisonCache()
+    }
+
+    /// We need to poison the cache when we start a given test as we share
+    private func poisonCache() throws {
+        let filePath = try XCTUnwrap(FileHandler.shared.glob(fixturePath, glob: "**/*.swift").first)
+        var contents = try FileHandler.shared.readTextFile(filePath)
+        contents += "\n"
+        try FileHandler.shared.write(contents, path: filePath, atomically: true)
+    }
+
+    public func run(_ command: (some AsyncParsableCommand).Type, _ arguments: [String] = []) async throws {
+        var arguments = arguments
+
+        if String(describing: command) == "GenerateCommand" {
+            arguments.append("--no-open")
+        }
+
+        if String(describing: command) == "TestCommand"
+            || String(describing: command) == "BuildCommand"
+        {
+            arguments.append(contentsOf: ["--derived-data-path", derivedDataPath.pathString])
+        }
```

Comment:
I'd have different functions:

```swift
func runTuistGenerate(open: Boolean, arguments: [String] =[])
func runTuistBuild(arguments: [String] = [])
func runTuist(_ command: (some AsyncParsableCommand).Type, _ arguments: [String] = [])
```

I think it's more ergonomic, plus we don't end up with a `run` with many `if/else`'s internally that assume what the caller's intent is.

Reply:
> I think it's more ergonomic

I will keep the same API but make the methods more explicit to get rid of the `if-else`. I don't want to recreate the whole CLI for acceptance tests and I think it's better to be explicit about what you are running, so the acceptance tests still read as the actual commands.